1 feature am

algmironov opened the pull request:
Созданы модели User, Report, ReportsPicture 
Созданы репозитории USERS, REPORTS, PICTURES
Созданы таблицы users, reports, pictures
Подключен liquibase + migrations

(application.properties не включен в коммит, тем не менее необходимые изменения имеются в aplication-team.properties , касающиеся БД и liquibase)

Kulachenkov committed:
Стартовый код

algmironov committed:
Созданы модели User, Report, ReportsPicture
Созданы репозитории USERS, REPORTS, PICTURES
Созданы таблицы users, reports, pictures
Подключен liquibase + migrations

algmironov committed:
Добавлены конструкторы, геттеры и сеттеры.
В модели Report время отправки отчета фиксируется самостоятельно, без внешнего участия - методом LocalDateTime.now()  в конструкторе.

algmironov commented:
отдельным коммитом добавил геттеры, сеттеры и конструкторы (рукалицо, забыл)

Delegant commented on target/classes/liquibase/scripts/migrations.sql:
```diff
@@ -0,0 +1,34 @@
+    name                 varchar(255) NOT NULL,
+    role                 varchar(255) NOT NULL DEFAULT 'USER'
+);
+
```
alter table users
    owner to "username";

Delegant commented on target/classes/liquibase/scripts/migrations.sql:
```diff
@@ -0,0 +1,34 @@
+    reportUpdateDate    timestamp,
+    user_id             bigint NOT NULL REFERENCES users(id)
+);
+
```
alter table reports
owner to "username";

Delegant commented on target/classes/liquibase/scripts/migrations.sql:
```diff
@@ -0,0 +1,34 @@
+    mediaType   text NOT NULL,
+    data        bytea NOT NULL,
+    report_id   bigint NOT NULL REFERENCES reports(id)
+);
```
alter table pictures
owner to "username";

Delegant reviewed (changes requested):
посмотрел, все отлично, но без назначения владельца таблицам не работает. 
liquibase вообще никак не стартует.

Kulachenkov committed:
Добавлены методы для создания клавиатуры и меню.
Можно создавать меню c кнопками и сообщением, передав параметры из статического класса

algmironov committed:
Merge remote-tracking branch 'origin/1-feature-NK' into 1_feature_AM

algmironov committed:
Updated TGBot version

Delegant reviewed (approved):
Все ок
